src/decide.ts

```diff
@@ -1,17 +1,83 @@
 import { PostHog } from './posthog-core'
-import { Compression, DecideResponse } from './types'
+import { Compression, DecideResponse, RemoteConfig } from './types'
 import { STORED_GROUP_PROPERTIES_KEY, STORED_PERSON_PROPERTIES_KEY } from './constants'
 
 import { logger } from './utils/logger'
-import { document } from './utils/globals'
+import { assignableWindow, document } from './utils/globals'
 
 export class Decide {
     constructor(private readonly instance: PostHog) {
         // don't need to wait for `decide` to return if flags were provided on initialisation
         this.instance.decideEndpointWasHit = this.instance._hasBootstrappedFeatureFlags()
     }
 
+    private _loadRemoteConfigJs(cb: (config?: RemoteConfig) => void): void {
+        if (assignableWindow.__PosthogExtensions__?.loadExternalDependency) {
+            assignableWindow.__PosthogExtensions__?.loadExternalDependency?.(this.instance, 'remote-config', () => {
+                return cb(assignableWindow._POSTHOG_CONFIG)
+            })
+        } else {
+            logger.error('PostHog Extensions not found. Cannot load remote config.')
+            cb()
+        }
+    }
+
+    private _loadRemoteConfigJSON(cb: (config?: RemoteConfig) => void): void {
+        this.instance._send_request({
+            method: 'GET',
+            url: this.instance.requestRouter.endpointFor('assets', `/array/${this.instance.config.token}/config`),
+            callback: (response) => {
+                cb(response.json as RemoteConfig | undefined)
+            },
```

Comment:
i guess we need to handle errors here?

Reply:
I guess but on the other hand this is the same as the current behaviour. If it fails, there's not a whole lot we can do about it. Same as if decide fails.